### ISSUE TITLE
Release and register as Julia package?

### ISSUE THREAD

**Issue by mtanneau** [JuliaSmoothOptimizers/QPSReader.jl#11]
Several Julia packages need to read MPS files, so having a stand-alone Julia reader for MPS files would avoid duplicating code.
For instance:
* [MathOptInterface.jl](https://github.com/JuliaOpt/MathOptInterface.jl) has an MPS reader/writer
* [Tulip.jl](https://github.com/ds4dm/Tulip.jl) has its own MPS reader (that handles free MPS format)
* [SDDP.jl](https://github.com/odow/SDDP.jl/issues/241) needs to parse MPS files when reading stochastic mps problem (I'm currently writing a parser for smps files and I'd like to avoid re-writing an MPS reader for the 3rd time)

Couple of questions:
* Would there be license restrictions when using QPSReader within another Julia package? (e.g., is it OK to use QPSReader within an MIT-licensed package?)
* For various reasons (especially in Tulip), I need constraint/objective matrices in COO format, not SparseCSC, so I would very much like to remove the COO -> SparseCSC conversion after parsing
* Performance is not (yet) an issue, but the reader in Tulip is ~2-3x faster than QPSReader

**Comment by dpo** [comment]
Yes, it would be useful to register this package. I'm not sure if other packages out there also read QPS.

> Would there be license restrictions when using QPSReader within another Julia package? (e.g., is it OK to use QPSReader within an MIT-licensed package?)

Question 13 [here](https://www.mozilla.org/en-US/MPL/2.0/FAQ/) says that you can combine MPL and BSD. I'm not sure about MIT. Question 14 suggests you can also combine with (L)GPL work.

> For various reasons (especially in Tulip), I need constraint/objective matrices in COO format, not SparseCSC, so I would very much like to remove the COO -> SparseCSC conversion after parsing

That would be a great addition!

> Performance is not (yet) an issue, but the reader in Tulip is ~2-3x faster than QPSReader

I admit I haven't profiled this package. I'm sure it can be improved. A pull request with benchmarks would be very welcome!

Thanks!

**Comment by odow** [comment]
> is it OK to use QPSReader within an MIT-licensed package

You can pull it in as a dependency that the user (or the Julia package manager) is responsible for installing. What you can't do is copy QPSReader verbatim, shove it into your repository, and then relicense under MIT. 

Essentially, with Julia the MIT only relates to the files that you are writing.

**Comment by mtanneau** [comment]
Can we tag and register the package once #21 is merged?
This is the last item I need to use it in Tulip.jl

**Comment by dpo** [comment]
Want to submit a PR with a version number in Project.toml?

**Comment by dpo** [comment]
https://github.com/JuliaRegistries/General/pull/12061
